flink-staging/flink-table/src/main/scala/org/apache/flink/api/table/expressions/analysis/ResolveFieldReferences.scala

```diff
@@ -30,21 +33,43 @@ import org.apache.flink.api.table.trees.Rule
  * Rule that resolved field references. This rule verifies that field references point to existing
  * fields of the input operation and creates [[ResolvedFieldReference]]s that hold the field
  * [[TypeInformation]] in addition to the field name.
+ * 
+ * @param inputOperation is optional but required if resolved fields should be pushed to underlying
+ *                       table sources.
+ * @param filtering defines if the field is resolved as a part of filtering operation or not
  */
-class ResolveFieldReferences(inputFields: Seq[(String, TypeInformation[_])])
+class ResolveFieldReferences(inputFields: Seq[(String, TypeInformation[_])],
+                             inputOperation: PlanNode,
+                             filtering: Boolean)
```

Comment:
It is Scala Style to indent the parameters by 4 spaces (even the first parameter) if they don't fit on a line, like so:
```scala
class ResolveFieldReferences(
    inputFields: Seq[(String, TypeInformation[_])],
    inputOperation: PlanNode,
    filtering: Boolean)  extends Rule[Expression] {
```